tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -70,12 +72,21 @@ object Build {
         linked
       }
 
+      implicit var incCompilationContext: IncCompilationContext = null
```

Comment:
Having an implicit null is never a good idea. Consider creating empty IncCompilationContext always, but run side-effecting methods like `collectFromPrev` explicitly on it's instance instead of making it a part of the constructor